File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2177.cpp
#include <bits/stdc++.h>
#define ll long long

using namespace std;

ll dist(ll xa, ll ya, ll xb, ll yb){ // retorna distancia ao quadrado
    return (xa-xb)*(xa-xb) + (ya-yb)*(ya-yb);
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    ll xi, yi, n, x, y, t;
    cin >> xi >> yi >> n;
    vector<ll> jogos;
    for(ll i = 0; i < n; i++){
        cin >> x >> y >> t;
        
        ll ans = dist(xi, yi, x, y);
        
        if(ans < t*t)
            jogos.push_back(i+1);
    }
    
    if(jogos.size() == 0) cout << -1 << endl;
    else{
        for(ll i = 0; i < jogos.size()-1; i++){
            cout << jogos[i] << " ";
        }
        cout << jogos[jogos.size()-1] << endl;
    }
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2177.py
def dist(xa, ya, xb, yb):
    return (xa-xb)**2 + (ya-yb)**2

ls = [int(i) for i in input().split()]

x = ls[0]
y = ls[1]
n = ls[2]

c = False
for i in range(n):
    lls = [int(j) for j in input().split()]

    xi = lls[0]
    yi = lls[1]
    ti = lls[2]
    
    ans = dist(xi, yi, x, y)
    
    if ans < ti**2:
        if c:
            print("", i+1, end = '')
        elif not c:    
            c = True
            print(i+1, end = '')
            
            
if not c:
    print(-1)